tabular/src/autogluon/tabular/models/widedeep_nn/callbacks.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import time
+from typing import Optional, Dict
+
+from pytorch_widedeep.callbacks import EarlyStopping
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: add time limit
+# TODO: refit_all support
+class EarlyStoppingCallbackWithTimeLimit(EarlyStopping):
+
+    def __init__(self, monitor: str = "val_loss", min_delta: float = 0.0, patience: int = 10, verbose: int = 0, mode: str = "auto",
+                 baseline: Optional[float] = None, restore_best_weights: bool = False,
+                 time_limit: Optional[int] = None, best_epoch_stop: Optional[int] = None):
+        super().__init__(monitor, min_delta, patience, verbose, mode, baseline, restore_best_weights)
+        self.best_epoch_stop = best_epoch_stop
+        self.time_limit = time_limit
+        self.start_time = time.time()
```

Comment:
better to have start_time as a parameter (this is done in LightGBM callback), that way we can re-use the time_limit and time.time found at the start of `_fit`.

Reply:
fixed